torch/_dynamo/symbolic_convert.py

```diff
@@ -148,6 +149,13 @@ def fail_and_restart_analysis(self):
         self.failed = True
         raise exc.SpeculationRestartAnalysis()
 
+    def mark_sound_and_restart_analysis(self):
```

Comment:
Won't restarting analysis in the success case will lead to O(n^2) tracing time?  This seems really bad.

Reply:
I mean it's O(n*m) where m is the number of autograd functions in your model, which I think isn't usually very large. Not sure if we want to chance that though. I can do a benchmark run to see if there's a slowdown? wdyt?

If this isn't acceptable I can change this to lighterweight correct checkpointing that likely only needs to checkpoint side effects because of the way HOPs are designed.

Reply:
`O(n*m)` is `O(n^2)`, since `m` can be large.  A simple example would be calling an autograd.Function in a loop.

What values actually need to get checkpointed here?

Reply:
So based on what we do in the other HOPs I think it's just side effects which can be globally accessed, because we throw away the inlining translator and subgraph tracer. I'll see if Voz has seen other weird things in autograd backward though just in case. It might be okay to only checkpoint side effects. We would accumulate extra guards though from backward which feels not nice, but is probably fine.

Reply:
As long as the HOP preserves behavior and protects against side effects, it should be fine. The restore logic predates this being a HOP.